lib/util.js

```diff
@@ -0,0 +1,170 @@
+    }
+  }
+}
+
+// The swagger standard does not accept the url param with ':'
+// so '/user/:id' is not valid.
+// This function converts the url in a swagger compliant url string
+// => '/user/{id}'
+function formatParamUrl (url) {
+  let start = url.indexOf('/:')
+  if (start === -1) return url
+
+  const end = url.indexOf('/', ++start)
+
+  if (end === -1) {
+    return url.slice(0, start) + '{' + url.slice(++start) + '}'
+  } else {
+    return formatParamUrl(url.slice(0, start) + '{' + url.slice(++start, end) + '}' + url.slice(end))
+  }
+}
+
+function consumesFormOnly (schema) {
+  const consumes = schema.consumes
+  return (
+    consumes &&
+      consumes.length === 1 &&
+      (consumes[0] === 'application/x-www-form-urlencoded' ||
+        consumes[0] === 'multipart/form-data')
+  )
+}
+
+// For supported keys read:
+// https://swagger.io/docs/specification/2-0/describing-parameters/
+function plainJsonObjectToSwagger2 (container, jsonSchema, externalSchemas) {
+  const obj = localRefResolve(jsonSchema, externalSchemas)
+  let toSwaggerProp
+  switch (container) {
+    case 'query':
+      toSwaggerProp = function (properyName, jsonSchemaElement) {
+        jsonSchemaElement.in = container
+        jsonSchemaElement.name = properyName
+        return jsonSchemaElement
+      }
+      break
+    case 'formData':
+      toSwaggerProp = function (properyName, jsonSchemaElement) {
+        delete jsonSchemaElement.$id
+        jsonSchemaElement.in = container
+        jsonSchemaElement.name = properyName
+
+        // https://json-schema.org/understanding-json-schema/reference/non_json_data.html#contentencoding
+        if (jsonSchemaElement.contentEncoding === 'binary') {
+          delete jsonSchemaElement.contentEncoding // Must be removed
+          jsonSchemaElement.type = 'file'
+        }
+
+        return jsonSchemaElement
+      }
+      break
+    case 'path':
+      toSwaggerProp = function (properyName, jsonSchemaElement) {
+        jsonSchemaElement.in = container
+        jsonSchemaElement.name = properyName
+        jsonSchemaElement.required = true
+        return jsonSchemaElement
+      }
+      break
+    case 'header':
+      toSwaggerProp = function (properyName, jsonSchemaElement) {
+        return {
+          in: 'header',
+          name: properyName,
+          required: jsonSchemaElement.required,
+          description: jsonSchemaElement.description,
+          type: jsonSchemaElement.type
+        }
+      }
+      break
+  }
+
+  return Object.keys(obj).reduce((acc, propKey) => {
+    acc.push(toSwaggerProp(propKey, obj[propKey]))
+    return acc
+  }, [])
+}
+
+function swagger2ParametersToOpenapi3 (jsonSchema) {
+  jsonSchema.schema = {}
+  jsonSchema.schema.type = jsonSchema.type
+  if (jsonSchema.type === 'object') {
+    jsonSchema.schema.properties = jsonSchema.properties
+  }
+  if (jsonSchema.type === 'array') {
+    jsonSchema.schema.items = jsonSchema.items
+  }
+  delete jsonSchema.type
```

Comment:
`delete` is an expensive operation, any reason why setting to undefined wouldn't suffice here?

Reply:
```delete``` is a must in this case, because swagger will check if the schema have invalid property or not. Setting to undefined will fail the test.